src/extension.c

```diff
@@ -15,12 +15,51 @@
 #include "query.h"
 #include <err.h>
 
+#define MAX_NAME_LEN 1024
+
 /* The registry for query expanders. Initialized by Extensions_Init() */
 static TrieMap *queryExpanders_g = NULL;
 
 /* The registry for scorers. Initialized by Extensions_Init() */
 static TrieMap *scorers_g = NULL;
 
+/**
+ * @brief Converts a string to uppercase.
+ *
+ * This function takes an input string and converts all its characters to uppercase.
+ * The converted string is stored in the output buffer. The function ensures that
+ * the output string is null-terminated.
+ *
+ * @param[in] in The input string to be converted to uppercase.
+ * @param[out] out The output buffer where the uppercase string will be stored.
+ * @param[in] inLen The length of the input string. If inLen is less than 0, the function
+ *                  calculates the length of the input string using strlen().
+ * @param[in] maxLen The maximum length of the output buffer. The function will not write
+ *                   more than maxLen characters to the output buffer, including the null terminator.
+ *
+ * @note The output buffer must be large enough to hold the converted string and the null terminator.
+ *       If the input string is longer than maxLen, only the first maxLen-1 characters will be converted
+ *       and stored in the output buffer, followed by a null terminator.
+ *
+ * @example
+ * @code
+ * char input[] = "Hello, World!";
+ * char output[50];
+ * str_toUpper(input, output, strlen(input), sizeof(output));
+ * printf("Uppercase: %s\n", output); // Output: "HELLO, WORLD!"
+ * @endcode
+ */
+static inline void str_toUpper(const char *in, char *out, size_t inLen, size_t maxLen) {
+  if (inLen < 0) {
```

Comment:
Isn't type of `inLen` unsigned (`size_t`)? Is this reachable?

Reply:
If `str_toUpper` is currently only used here and always gets a valid `inLen` consider dropping this if block?

Reply:
Yeah I wanted it for generality but there's no real need..
Will modify it